src/websocket_endpoint.cpp

```diff
@@ -76,6 +82,30 @@ namespace sensr
     err_receiver_ = 0;
   }
 
+  context_ptr WebSocketEndPoint::OnTSLInit(websocketpp::connection_hdl hdl) {
+    context_ptr ctx = websocketpp::lib::make_shared<boost::asio::ssl::context>(boost::asio::ssl::context::sslv23);
+
+    try {
+        ctx->set_options(boost::asio::ssl::context::default_workarounds |
+                         boost::asio::ssl::context::no_sslv2 |
+                         boost::asio::ssl::context::no_sslv3 |
+                         boost::asio::ssl::context::single_dh_use);
+
+
+        ctx->set_verify_mode(boost::asio::ssl::verify_peer);
+        ctx->set_verify_callback(std::bind(&WebSocketEndPoint::OnVerifyCertificate, 
+          this,
+          std::placeholders::_1, 
+          std::placeholders::_2));
+
+        // Here we load the CA certificates of all CA's that this client trusts.
+        ctx->load_verify_file(cert_path_);
+    } catch (std::exception& e) {
+        std::cout << e.what() << std::endl;
+    }
+    return ctx;
```

Comment:
Is it fine to return this if the exception is raised?  Maybe we should change it to `nullptr`
```suggestion
    } catch (std::exception& e) {
        std::cout << e.what() << std::endl;
        ctx = nullptr;
    }
    return ctx;
```

Reply:
let me try to do it.

Reply:
I think this change made the fix of https://seoulrobotics.atlassian.net/browse/AR-2853